domain/domain.go

```diff
@@ -490,12 +502,21 @@ func (do *Domain) Reload() error {
 		return err
 	}
 
-	is, hitCache, oldSchemaVersion, changes, err := do.loadInfoSchema(ver.Ver)
-	metrics.LoadSchemaDuration.Observe(time.Since(startTime).Seconds())
+	version := ver.Ver
+	is, hitCache, oldSchemaVersion, changes, err := do.loadInfoSchema(version)
 	if err != nil {
-		metrics.LoadSchemaCounter.WithLabelValues("failed").Inc()
-		return err
+		if version = getFlashbackStartTSFromErrorMsg(err); version != 0 {
+			// use the lastest available version to create domain
+			version -= 1
+			is, hitCache, oldSchemaVersion, changes, err = do.loadInfoSchema(version)
+		}
+		if err != nil {
+			metrics.LoadSchemaDuration.Observe(time.Since(startTime).Seconds())
+			metrics.LoadSchemaCounter.WithLabelValues("failed").Inc()
+			return err
+		}
 	}
```

Comment:
You could move L518 to L512. I prefer leaving the metrics block at the top indent:

```
xx := load(ver.Ver)
if err != nil {
   // retry with flashback
   xx = load(version - 1)
}
if err != nil {
   return err
}
```

Reply:
Done, PTAL @xhebox